src/nav_msg.c

```diff
@@ -73,6 +73,36 @@ static u32 extract_word(nav_msg_t *n, u16 bit_index, u8 n_bits, u8 invert)
   return word >> (32 - n_bits);
 }
 
+static void update_bit_sync(nav_msg_t *n, s32 corr_prompt_real)
+{
+  float dot = corr_prompt_real * n->prev_corr;
+  n->prev_corr = corr_prompt_real;
+
+  if (dot > 0)
+    return;
+
+  /* Sign change - Add to histogram */
+  n->hist[n->bit_phase] += -dot;
+
+  n->bit_phase_count++;
```

Comment:
Add comment to explain this waits until we have seen `NAV_MSG_BIT_PHASE_THRES` sign flips before looking at the histogram